source/MaterialXFormat/Util.cpp

```diff
@@ -164,6 +164,15 @@ void flattenFilenames(DocumentPtr doc, const FileSearchPath& searchPath, StringR
         }
         string resolvedString = valueElem->getResolvedValueString(elementResolver);
 
+        if (isRelative)
+        {
+            FilePath resolvedValue(resolvedString);
+            if (isRelative(resolvedValue, relativeTexturePath))
```

Comment:
Check to see if the resolved string is relative to a texture path, if it is skip it (we don't want to make the path absolute).